packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -1438,17 +1458,48 @@ export class IntervalCollection<TInterval extends ISerializableInterval>
             this.client.rebasePosition(end, sequenceNumber, localSeq);
 
         const intervalId = properties?.[reservedIntervalIdKey];
+        const localInterval = this.localCollection.getIntervalById(intervalId);
+
         const rebased: ISerializedInterval = {
             start: startRebased,
             end: endRebased,
             intervalType,
             sequenceNumber: this.client?.getCurrentSeq() ?? 0,
             properties,
         };
+
         if (opName === "change" && (this.hasPendingChangeStart(intervalId) || this.hasPendingChangeEnd(intervalId))) {
             this.removePendingChange(serializedInterval);
             this.addPendingChange(intervalId, rebased);
         }
+
+        if (!localInterval) {
+            return rebased;
+        }
+
+        // TODO: message
+        assert(localInterval instanceof SequenceInterval, "");
```

Comment:
I wasn't entirely sure how to articulate what we are verifying in this assertion. Reviewers:
1. Is this assertion ok to make?
2. What is the purpose of such an assertion?

Reply:
we do an early exit when `this.client` is undefined above, and this class is used with SequenceIntervals iff `this.client` exists, so an assert here seems fine. Just casting is potentially reasonable with a note to that effect (asserts make more sense for complex state machines--here it's not too hard to see from current code structure why the assert is true, and I don't think it's that likely we'll change this area much).